javascript/transform.js

```diff
@@ -15,27 +16,30 @@ function outOfChina(lat, lng) {
 	return false;
 }
 
-function transformLat(x, y) {
-	var ret = -100.0 + 2.0*x + 3.0*y + 0.2*y*y + 0.1*x*y + 0.2*Math.sqrt(Math.abs(x));
-	ret += (20.0*Math.sin(6.0*x*Math.PI) + 20.0*Math.sin(2.0*x*Math.PI)) * 2.0 / 3.0;
-	ret += (20.0*Math.sin(y*Math.PI) + 40.0*Math.sin(y/3.0*Math.PI)) * 2.0 / 3.0;
-	ret += (160.0*Math.sin(y/12.0*Math.PI) + 320*Math.sin(y*Math.PI/30.0)) * 2.0 / 3.0;
+function gcjTrigLat(x, y) {
+	var ret = 20.0*Math.sin(6.0*x*Math.PI) + 20.0*Math.sin(2.0*x*Math.PI);
+	ret += 20.0*Math.sin(y*Math.PI) + 40.0*Math.sin(y/3.0*Math.PI);
+	ret += 160.0*Math.sin(y/12.0*Math.PI) + 320*Math.sin(y*Math.PI/30.0);
+	ret *= 2.0 / 3.0;
+	ret += -100.0 + 2.0*x + 3.0*y + 0.2*y*y + 0.1*x*y + 0.2*Math.sqrt(Math.abs(x));
 	return ret;
 }
 
-function transformLon(x, y) {
-	var ret = 300.0 + x + 2.0*y + 0.1*x*x + 0.1*x*y + 0.1*Math.sqrt(Math.abs(x));
-	ret += (20.0*Math.sin(6.0*x*Math.PI) + 20.0*Math.sin(2.0*x*Math.PI)) * 2.0 / 3.0;
```

Comment:
The most terrible part of writing the long expr this way is that compilers will not optimize this section to respect IEEE754…